src/Microsoft.ML.Data/DataLoadSave/Database/DatabaseSource.cs

```diff
@@ -25,6 +25,26 @@ public DatabaseSource(DbProviderFactory providerFactory, string connectionString
             CommandText = commandText;
         }
 
+        /// <summary>Creates a new instance of the <see cref="DatabaseSource" /> class.</summary>
+        /// <param name="providerFactory">The factory used to create the <see cref="DbConnection"/>..</param>
+        /// <param name="connectionString">The string used to open the connection.</param>
+        /// <param name="commandText">The text command to run against the data source.</param>
+        /// <param name="commandTimeOut">The timeout(in seconds) for database command.</param>
+        public DatabaseSource(DbProviderFactory providerFactory, string connectionString, string commandText, int commandTimeOut)
+        {
+            Contracts.CheckValue(providerFactory, nameof(providerFactory));
+            Contracts.CheckValue(connectionString, nameof(connectionString));
+            Contracts.CheckValue(commandText, nameof(commandText));
+
+            ProviderFactory = providerFactory;
+            ConnectionString = connectionString;
+            CommandText = commandText;
+            CommandTimeOut = commandTimeOut;
+        }
+
+        /// <summary>Gets the timeout for database command.</summary>
+        public int? CommandTimeOut { get; }
```

Comment:
**Suggestion**: Change this property name to be `CommandTimeOutInSeconds`, as this was recommended on the original PR by @tannergooding (whom I think introduced the DbLoader into ML.NET)

https://github.com/dotnet/machinelearning/pull/4494#discussion_r349254631